Add Hack for reading more sensors

JoergOstertag opened the pull request:
On my iMac late 2011 i replaced the broken HDD with a new SSD (Without Apple Thermal Sensor).
After this I adapted the part of temperature reading to also read all Temperature-Sensors under /sys/devices/platform/applesmc.768 too.
In addition I printed out the Sensor Labels for easier debugging.

gaul reviewed (changes requested):
This substantially breaks my MBA 2014 setup.  Before:

```
Old Temp: 64 New Temp: 64 Fan: Exhaust Speed: 1693
```

After:

```
Old Temp: 35099 New Temp: 35099 Fan: Exhaust Speed: 6500
```

This PR has too many unrelated changes.  Please narrow it and submit individual PRs for each one.

gaul commented on src/mbpfan.c:
```diff
@@ -211,12 +211,93 @@ t_sensors *retrieve_sensors()
+		fscanf(file, "%s", label);
+		fclose(file);
+		if (label[1] != 'G' ){ // GPU Only
+		  //		  continue;
```
Dead code.

JoergOstertag replied:
You might Try if commenting this in would get it working again with your Macbook?
I think later someone can introduce a Black/Whitelist for sensors.
I one wants to introduce Black/Whitelist it would be nice to use labeel Names for the Configuration. This was another reason to introduce named Labels.

gaul replied:
Sorry I don't have time to experiment with this.  I recommend narrowing this PR to minimize the chance of regression, even if that requires several PRs and more work for you.

gaul commented on src/mbpfan.c:
```diff
@@ -135,7 +135,7 @@ t_sensors *retrieve_sensors()
     char *path = NULL;
     char *path_begin = NULL;
-
+    
```
Whitespace change.

JoergOstertag replied:
Can you give me a hint which linter to use for auto formating? Or do you have a link to emacs/eclipse settings?

gaul commented on src/mbpfan.c:
```diff
@@ -211,12 +211,93 @@ t_sensors *retrieve_sensors()
+		}
+
+		strncpy(labelType, "???",80);
+		if (label[1] == 'A' ){ strncpy(labelType, "Ambient", 80); }
```
Why do you need to copy these?

JoergOstertag replied:
Idea was to later also copy it to s_sensors. So I can later display which sensor triggered the max Temperature.

gaul replied:
But you can just set `labelType = "Ambient"`?

gaul commented on src/mbpfan.c:
```diff
@@ -211,12 +211,93 @@ t_sensors *retrieve_sensors()
+	      free(path);
+	      if(NULL == file) {
+		continue;
+	      } else {
```
Deindent.

JoergOstertag commented:
> On my iMac late 2011 i replaced the broken HDD with a new SSD (Without Apple Thermal Sensor).
> After this I adapted the part of temperature reading to also read all Temperature-Sensors under /sys/devices/platform/applesmc.768 too.
> In addition I printed out the Sensor Labels for easier debugging.



> This substantially breaks my MBA 2014 setup. Before:
> 
> ```
> Old Temp: 64 New Temp: 64 Fan: Exhaust Speed: 1693
> ```
> 
> After:
> 
> ```
> Old Temp: 35099 New Temp: 35099 Fan: Exhaust Speed: 6500
> ```

I already read that some Macbooks have temperature sensors with really high of random values. To see more here  and track down your problem with the high sensor reading I added Labels to the debug output. Now we should be able to see which sensor we might have to exclude.
You can try the Version directly at: https://github.com/JoergOstertag/mbpfan

> 
> This PR has too many unrelated changes. Please narrow it and submit individual PRs for each one.

Concerning splitting the pull requests: Up to now I only used straight forward pushes; I am glad I got this pull request on the road ;-)
But I think one thing is important before I continue here: Find out which code-formatter you use. Otherwise I'll go crazy with reformatting.

gaul commented:
>  But I think one thing is important before I continue here: Find out which code-formatter you use. Otherwise I'll go crazy with reformatting.

Unfortunately this project does not have a consistent formatting (I am not the original maintainer, nor is the previous maintainer the original one).  It would be nice to make it more uniform via `clang-format` although this would require some trial-and-error to find a minimal configuration that mimics what exists.  If you want to take this up let's do it as a separate PR.

JoergOstertag commented:
> > But I think one thing is important before I continue here: Find out which code-formatter you use. Otherwise I'll go crazy with reformatting.
> 
> Unfortunately this project does not have a consistent formatting (I am not the original maintainer, nor is the previous maintainer the original one). It would be nice to make it more uniform via `clang-format` although this would require some trial-and-error to find a minimal configuration that mimics what exists. If you want to take this up let's do it as a separate PR.

I personally would suggest to do a reformatting of the whole code to one of the commonly used standards. I would suggest to use either K&R (default build in Eclipse) or default clang-format. Yes this is a large change, but after this you should be save :-)

gaul commented:
> I personally would suggest to do a reformatting of the whole code to one of the commonly used standards. I would suggest to use either K&R (default build in Eclipse) or default clang-format. Yes this is a large change, but after this you should be save :-)

I understand style is important to some people but I maintain a bunch of projects and hacking out real functionality is what is important to me.  If you can minimize this PR and follow whatever inconsistent formatting you can reopen or open a new pull request.  As I said before, I will accept clang-tidy reformattings that stepwise move towards something consistent but I am not interesting in style changes intermingled with functional changes.
